allensdk/brain_observatory/behavior/stimulus_processing/__init__.py

```diff
@@ -152,12 +163,29 @@ def get_gratings_metadata(stimuli: Dict, start_idx: int = 0) -> pd.DataFrame:
     return grating_df
 
 
-def get_stimulus_templates(pkl) -> Optional[StimulusTemplate]:
+def get_stimulus_templates(pkl: dict,
+                           pkl_path: str,
```

Comment:
I don't like the idea of passing in the `pkl_path` just so it can be used to populate an error message. It leaves open the possibility for a future developer to pass in inconsistent `pkl` and `pkl_path` arguments. Should this method just accept `pkl_path` and call `pd.read_pickle` on its own? I know that means that the method and the `BehaviorSession` will both end up calling `pd.read_pickle`, but that seems pretty lightweight.

Alternatively, this method could just accept an object that inherits from `BehaviorDataTransforms`, ensuring that the input data is all self-consistent.

Reply:
Generally, I dislike mixing IO (reading and writing data) with processing data because it makes everything harder. An example are the current tests for this `get_stimulus_templates` function, where passing a `dict` is easy, but needing to write infrastructure to write a test `dict` to a `*.pkl` file just to be able to test the function is pretty gross.

I think what I will do is just remove the `pkl_path` param and deal with a slightly less informative warning message in the case that neither `images` nor `gratings` exist for a particular stimulus *.pkl.

Reply:
That sounds fine to me.